node/core/pvf/src/prepare/worker.rs

```diff
@@ -314,59 +307,61 @@ async fn recv_request(stream: &mut UnixStream) -> io::Result<(Vec<u8>, PathBuf,
 /// The entrypoint that the spawned prepare worker should start with. The `socket_path` specifies
 /// the path to the socket used to communicate with the host.
 pub fn worker_entrypoint(socket_path: &str) {
-	worker_event_loop("prepare", socket_path, |mut stream| async move {
+	worker_event_loop("prepare", socket_path, |stream| async move {
+		let mutex = Arc::new(Mutex::new((stream, false)));
 		loop {
-			let (code, dest, preparation_timeout) = recv_request(&mut stream).await?;
-
+			let (code, dest, preparation_timeout) = {
+				let mut lock = mutex.lock().await;
+				// Unset the lock flag. We set it when either thread finishes.
```

Comment:
Do you have to do this due to possible previous job? Would be nice to have a more descriptive comment about this